auto_install/install.sh

```diff
@@ -2482,52 +2667,61 @@ askUnattendedUpgrades(){
 
 confUnattendedUpgrades(){
 	local PIVPN_DEPS
-	PIVPN_DEPS=(unattended-upgrades)
-	installDependentPackages PIVPN_DEPS[@]
-	aptConfDir="/etc/apt/apt.conf.d"
 
-	if [ "$PLAT" = "Ubuntu" ]; then
+	if [[ "${PKG_MANAGER}" == 'apt-get' ]]; then
+		PIVPN_DEPS=(unattended-upgrades)
+		installDependentPackages PIVPN_DEPS[@]
+		aptConfDir="/etc/apt/apt.conf.d"
+
+		if [ "$PLAT" = "Ubuntu" ]; then
+			# Ubuntu 50unattended-upgrades should already just have security enabled
+			# so we just need to configure the 10periodic file
+			echo "APT::Periodic::Update-Package-Lists \"1\";
+		APT::Periodic::Download-Upgradeable-Packages \"1\";
+		APT::Periodic::AutocleanInterval \"5\";
+		APT::Periodic::Unattended-Upgrade \"1\";" | $SUDO tee "${aptConfDir}/10periodic" > /dev/null
+		else
 
-		# Ubuntu 50unattended-upgrades should already just have security enabled
-		# so we just need to configure the 10periodic file
-		echo "APT::Periodic::Update-Package-Lists \"1\";
-	APT::Periodic::Download-Upgradeable-Packages \"1\";
-	APT::Periodic::AutocleanInterval \"5\";
-	APT::Periodic::Unattended-Upgrade \"1\";" | $SUDO tee "${aptConfDir}/10periodic" > /dev/null
+			# Raspbian's unattended-upgrades package downloads Debian's config, so we copy over the proper config
+			# Source: https://github.com/mvo5/unattended-upgrades/blob/master/data/50unattended-upgrades.Raspbian
+			if [ "$PLAT" = "Raspbian" ]; then
+				$SUDO install -m 644 "${pivpnFilesDir}/files${aptConfDir}/50unattended-upgrades.Raspbian" "${aptConfDir}/50unattended-upgrades"
+			fi
 
-	else
+			# Add the remaining settings for all other distributions
+			echo "APT::Periodic::Enable \"1\";
+		APT::Periodic::Update-Package-Lists \"1\";
+		APT::Periodic::Download-Upgradeable-Packages \"1\";
+		APT::Periodic::Unattended-Upgrade \"1\";
+		APT::Periodic::AutocleanInterval \"7\";
+		APT::Periodic::Verbose \"0\";" | $SUDO tee "${aptConfDir}/02periodic" > /dev/null
+		fi
 
-		# Raspbian's unattended-upgrades package downloads Debian's config, so we copy over the proper config
-		# Source: https://github.com/mvo5/unattended-upgrades/blob/master/data/50unattended-upgrades.Raspbian
-		if [ "$PLAT" = "Raspbian" ]; then
-			$SUDO install -m 644 "${pivpnFilesDir}/files${aptConfDir}/50unattended-upgrades.Raspbian" "${aptConfDir}/50unattended-upgrades"
+		# Enable automatic updates via the bullseye repository when installing from debian package
+		if [ "$VPN" = "wireguard" ]; then
+			if [ -f /etc/apt/sources.list.d/pivpn-bullseye-repo.list ]; then
+				if ! grep -q "\"o=$PLAT,n=bullseye\";" "${aptConfDir}/50unattended-upgrades"; then
+					$SUDO sed -i "/Unattended-Upgrade::Origins-Pattern {/a\"o=$PLAT,n=bullseye\";" "${aptConfDir}/50unattended-upgrades"
+				fi
+			fi
 		fi
+	elif [[ "${PKG_MANAGER}" == 'apk' ]]; then
+		echo 'https://dl-cdn.alpinelinux.org/alpine/edge/testing/' >> /etc/apk/repositories && apk -q update
```

Comment:
Missing sudo when writing to `/etc/apk/repositories` and running `apk update`. Anyway why are you enabling the [edge repository](https://wiki.alpinelinux.org/wiki/Upgrading_Alpine#Upgrading_to_Edge)?

Reply:
To install `apk-autoupdate`

Reply:
But edge is a development branch, users may not expect that PiVPN makes this significant change. Also the wiki explicitly recommends agains having both stable and edge enabled at the same time:

> Warning: Do not enable stable and edge repos at the same time. This can break your system. Either use edge or stable.

Reply:
I tested this change and adding this repo and downloading `apk-autoupdate` doesn't break the Docker Image
If you prefer, I can remove the repo after downloading the package so the change is not permanent

Reply:
> I tested this change and adding this repo and downloading `apk-autoupdate` doesn't break the Docker Image If you prefer, I can remove the repo after downloading the package so the change is not permanent

I have tested it and it is doable
The only problem is that if you upgrade packages with the `prune` option (`apk upgrade --prune`) then `apk-autoupdate` will be removed

Reply:
I'm not sure about the meaning of _This can break your system_ but I assume it will cause issue when installing new packages or upgrading existing.

> I tested this change and adding this repo and downloading apk-autoupdate doesn't break the Docker Image
If you prefer

Because it is a single package not available in other repositories.

Generally you don't upgrade packages in the container itself, you just deploy a new container. So unattended upgrades don't make much sense in a container, where the root file system could be read only too.

> I can remove the repo after downloading the package so the change is not permanent

Yeah maybe, a bit ugly to see though...

Reply:
> Yeah maybe, a bit ugly to see though...

And how do we go about the `UNATTUPG` flag?

Reply:
What do you mean?

Reply:
> The only problem is that if you upgrade packages with the `prune` option (`apk upgrade --prune`) then `apk-autoupdate` will be removed disabling unattended upgrades without switching the `UNATTUPG` flag

Reply:
> The only problem is that if you upgrade packages with the prune option (apk upgrade --prune) then apk-autoupdate

This is going to happen if you add the repository - install apk-autoupdate - remove the repository, right?

The `${UNATTUPG}` flag is just used at install time to save the fact than unattended upgrades need to be enabled and to support enabling them via unattended PiVPN installation.

Reply:
> This is going to happen if you add the repository - install apk-autoupdate - remove the repository, right?

Yep

> The `${UNATTUPG}` flag is just used at install time to save the fact than unattended upgrades need to be enabled and to support enabling them via unattended PiVPN installation.

Ok, so not a problem